examples/api-gateway-authorizer/infra/api.ts

```diff
@@ -0,0 +1,19 @@
+import { api, jwtAuthorizerConfig, router } from "@notation/aws/api-gateway";
+import { getUserHandler } from "runtime/user.fn";
+
+if (!process.env.ISSUER || !process.env.AUDIENCE) {
+  throw new Error(
+    "ISSUER and AUDIENCE environment variables must be set for the api-with-authorizer demo app",
+  );
+}
+
+const userApi = api({ name: "user-api" });
+const userApiRouter = router(userApi);
+
+const userAuthConfig = jwtAuthorizerConfig(
+  "authorizerTestTodos2",
+  process.env.ISSUER,
+  [process.env.AUDIENCE],
+);
+
+userApiRouter.get("/user", getUserHandler, userAuthConfig);
```

Comment:
Maybe cleaner to do something like:

```ts
userApiRouter.get("/user", {
  handler: getUserHandler,
  authorizer: { type: 'jwt', issuer, audience }, 
})
```

Where the second argument can be either a function or a config object.

Reply:
Ah just realised that you had it like this because you had two endpoints, before I deleted one of them.

Maybe the auth could be applied to the router?

```ts
const authenticatedUserRouter = router(userApi).auth({ type: "jwt", issuer, audience });
```

Or something like that!